Source/Charts/Formatters/DefaultValueFormatter.swift

```diff
@@ -22,69 +23,65 @@ open class DefaultValueFormatter: NSObject, ValueFormatter
     
     @objc open var block: Block?
     
-    @objc open var hasAutoDecimals: Bool = false
+    @objc open var hasAutoDecimals: Bool
     
-    private var _formatter: NumberFormatter?
-    @objc open var formatter: NumberFormatter?
-    {
-        get { return _formatter }
-        set
-        {
+    @objc open var formatter: NumberFormatter? {
+        willSet {
             hasAutoDecimals = false
-            _formatter = newValue
         }
     }
     
-    private var _decimals: Int?
-    open var decimals: Int?
-    {
-        get { return _decimals }
-        set
-        {
-            _decimals = newValue
-            
-            if let digits = newValue
+    open var decimals: Int? {
+        didSet {
+            if let digits = decimals
             {
-                self.formatter?.minimumFractionDigits = digits
-                self.formatter?.maximumFractionDigits = digits
-                self.formatter?.usesGroupingSeparator = true
+                formatter?.minimumFractionDigits = digits
+                formatter?.maximumFractionDigits = digits
+                formatter?.usesGroupingSeparator = true
             }
         }
     }
     
     public override init()
     {
-        super.init()
-        
-        self.formatter = NumberFormatter()
+        formatter = NumberFormatter()
+        formatter?.usesGroupingSeparator = true
```

Comment:
why set usesGroupingSeparator and decimals here while the old code does not? did I miss anything

Reply:
I made the empty initializer provide the same formatter that `ChartUtils.defaultFormatter` had which set it to true

Reply:
Oh sorry. I missed the removal